<?xml version="1.0" ?> 
<launch> 
  <!-- Define robot IP, load gripper, and publication rate arguments -->
  <arg name="robot_ip" value="10.224.20.198"/> 
  <arg name="load_gripper" value="false"/> 
  <!-- Define arm ID argument -->
  <arg name="arm_id" value="panda"/> 

  <!-- Include the franka_control launch file and pass all arguments -->
  <!--<include file="$(find franka_control)/launch/franka_control.launch" pass_all_args="true"> -->
  <include file="$(find joint_velocity_controller)/launch/control_node.launch" pass_all_args="true">
    <arg name="robot_ip" value="$(arg robot_ip)"/> 
    <arg name="load_gripper" value="$(arg load_gripper)"/> 
  </include>
  
  <!-- Load the load parameters -->
 <node name="set_load_client" pkg="joint_velocity_controller" type="set_load_client" output="screen">
    <param name="mass" value="0.553455" />
    <rosparam param="F_x_center_load"> [-0.010328, 0.000068, 0.148159] </rosparam>
    <rosparam param="load_inertia">  [0.02001, 0.000006527121, -0.0004590, 0.000006527121, 0.01936, 0.000003371038, -0.0004590, 0.000003371038, 0.002245] </rosparam>
  </node>  

  <!--<node name="set_load_client" pkg="joint_velocity_controller" type="set_load_client" output="screen">
    <param name="mass" value="0" />
    <rosparam param="F_x_center_load"> [-0, 0, 0] </rosparam>
    <rosparam param="load_inertia">  [0, 0, -0, 0, 0, 0, -0, 0, 0] </rosparam>
  </node> -->

  <!-- Load the joint group velocity controller parameters -->
  <rosparam command="load" file="$(find joint_velocity_controller)/config/default.yaml" subst_value="true" /> 
  
  <!-- Spawn the joint group velocity controller -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="joint_group_velocity_controller"/>
</launch>

